### ISSUE TITLE
Create a tool to summarise results

### ISSUE THREAD

**Issue by samelamin** [vercel/ai#2255]
### Description

I have a tool that returns a list of json objects and I am trying to create a tool to summarise the results

unfortunately the summarise results tool never gets called, I suspect its because I don't have an `execute` function, but I want this as an agent rather than the system prompt because this tool shouldn't always run, but rather run when required

Is this possible? 

### Code example

```
  analyseJsonResuls: tool({
    description: "This tool should run when we have json results and should give a brief summary",
    parameters: z.object({ queryResults: z.string().describe("Query results returned from the data store, the data will be in JSON format") }),
    // execute: async ({analysis}) => analysis,
  }),
```

### Additional context

The text field from the `generateText` is empty even though the tool that runs to get queryResults work

so the data is returned, but the LLm isn't kicking off the `analyseJsonResuls` tool

**Comment by lgrammel** [comment]
Are you using `maxToolRoundtrips`? Also, have you considered extending your tool that returns the json object to perform the summarization as well?

**Comment by samelamin** [comment]
Thanks for the quick response @lgrammel 

my `maxToolRoundtrips` is set to 0 because otherwise `toolResults` will be empty and I will have to manually extract the data from `responseMessages` which can be a pain lol

Extending it could work but then I will also have to manually call the llm with the results because the data is returned via code in the `execute` function. Or did I perhaps miss something obvious?

**Comment by lgrammel** [comment]
If you set `maxToolRountrips` to `0`, you need to call `generateText` a 2nd time with the assistant message (that contains the tool call) attached to the initial messages.

**Comment by samelamin** [comment]
Understood Is there a way to get more than 2 calls in the toolResults section or is this perhaps a bug?

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/issues/2176

**Comment by lgrammel** [comment]
Closing as duplicate of https://github.com/vercel/ai/issues/2176

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **None**  
Confidence: **High**

### Rationale
This is primarily a usage/support question followed by duplicate bookkeeping, without a clearly established failure of documented SDK behavior in the issue itself.

### Evidence
- Reporter asks how to trigger a summarization tool after another tool returns JSON results.
- Maintainer explains that with `maxToolRoundtrips: 0`, a second `generateText` call must be made manually.
- The follow-up about obtaining multiple calls in `toolResults` was closed as a duplicate of issue #2176.
